packages/@romejs/js-parser/parser/jsx.ts

```diff
@@ -32,7 +32,7 @@ import {isValidIdentifierName} from "@romejs/js-ast-utils";
 
 // Indicates whether we should create a JSXIdentifier or a JSXReferenceIdentifier
 function isHTMLTagName(tagName: string): boolean {
-	return /^[a-z]|-/.test(tagName) && isValidIdentifierName(tagName);
+	return /^[a-z0-9]+$|-/.test(tagName) && isValidIdentifierName(tagName);
```

Comment:
```suggestion
	return /^[a-z0-9]|-/.test(tagName) && isValidIdentifierName(tagName);
```

The original regular expression checked that the first letter was lowercase, and then checked that the identifier as a whole was valid. Changing this to check the entire string feels unnecessary and potentially brittle.

Reply:
I see what you're saying. this could make the test easier if I stick with the original regex and then abide by the[ rule ](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md) saying that it won't check it if the name begins with a lower case.